cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/HashLoginServiceApiCredentials.java

```diff
@@ -0,0 +1,333 @@
+                || apiUsers.getValueFrom().getSecretKeyRef().getKey() == null) {
+            throw new InvalidResourceException("The configuration of the Cruise Control REST API users " +
+                    "referenced in spec.cruiseControl.apiUsers is invalid.");
+        }
+    }
+
+    /**
+     * Parse String containing API credential config into map of API user entries.
+     *
+     * @param config API credential config file as a String.
+     *
+     * @return map of API credential entries containing username, password, and role.
+     */
+    /* test */ static Map<String, UserEntry> parseEntriesFromString(String config) {
+        Map<String, UserEntry> entries = new HashMap<>();
+
+        if (config.isEmpty()) {
+            return entries;
+        }
+
+        for (String line : config.split("\n")) {
+            Matcher matcher = HASH_LOGIN_SERVICE_PATTERN.matcher(line);
+            if (matcher.matches()) {
+                String[] parts = line.replaceAll("\\s", "").split(":");
+                String username = parts[0];
+                String password = parts[1].split(",")[0];
+                Role role = Role.fromString(parts[1].split(",")[1]);
+                if (entries.containsKey(username)) {
+                    throw new InvalidConfigurationException("Duplicate username found: " + "\"" + username + "\". "
+                            + "Cruise Control API credentials config must contain unique usernames");
+                }
+                entries.put(username, new UserEntry(username, password, role));
+            } else {
+                throw new InvalidConfigurationException("Invalid configuration provided: " +  "\"" + line + "\". " +
+                        "Cruise Control API credentials config must follow " +
+                        "HashLoginService's file format `username: password [,rolename ...]`");
+            }
+        }
+        return entries;
+    }
+
+    /**
+     * Returns Cruise Control API auth credentials file as a String.
+     *
+     * @param entries Map of API user entries containing API user credentials.
+     *
+     * @return Returns Cruise Control API auth credentials file as a String.
+     */
+    private static String generateApiAuthFileAsString(Map<String, UserEntry> entries) {
+        StringBuilder sb = new StringBuilder();
+        // Follows  Jetty's HashLoginService's file format: username: password [,rolename ...]
+        for (UserEntry e : entries.values()) {
+            sb.append(e.username).append(": ").append(e.password).append(", ").append(e.role()).append("\n");
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Parses auth data from existing Topic Operator API user secret into map of API user entries.
+     *
+     * @param secret API user secret
+     *
+     * @return Map of API user entries containing user-managed API user credentials
+     */
+    /* test */ static Map<String, UserEntry> generateToManagedApiCredentials(Secret secret) {
```

Comment:
Would `generateTOManagedApiCredentials` be better? It is a shortcut, so you can use both letters in capital. It is quite confusing here given `to` is also a common word used in method names.